pkg/process/handler.go

```diff
@@ -36,163 +36,123 @@ func (h *RegistrationHandler) GetDepartmentByID(c echo.Context) error {
 func (h *RegistrationHandler) GetRegistrationsByPatient(c echo.Context) error {
 
 	c.Logger().Debug("hello world")
 	return c.JSON(http.StatusCreated, api.Return("ok", nil))
 }
+// NOTE: use cookie for identification
 
 // GetRegistrationsByDoctor
 // @Summary get all registrations (doctor view)
 // @Tags doctor
 // @Description display all registrations of a patient
-// @Param doctor_id path uint true "Patient ID"
 // @Produce json
-// @Success 200 {array} api.ReturnedData{data=[]Registration}
-// @Router /doctor/{doctor_id}/registrations [GET]
+// @Success 200 {object} api.ReturnedData{data=[]Registration}
+// @Router /doctor/registrations [GET]
 func (h *RegistrationHandler) GetRegistrationsByDoctor(c echo.Context) error {
 
 	c.Logger().Debug("hello world")
 	return c.JSON(http.StatusCreated, api.Return("ok", nil))
 }
+// NOTE: use cookie for identification
 
-// GetRegistrationDetailByPatient
+
+// GetRegistrationByPatient
 // @Summary get a registration by its ID (patient view)
 // @Tags patient
 // @Description return a registration details by its ID
-// @Param registration_id path uint true "Registration ID"
+// @Param RegistrationID body uint true "registration's ID"
 // @Produce json
 // @Success 200 {object} api.ReturnedData{data=Registration}
-// @Router /patient/registration/{registration_id} [GET]
-func (h *RegistrationHandler) GetRegistrationDetailByPatient(c echo.Context) error {
+// @Router /patient/registration [POST]
+func (h *RegistrationHandler) GetRegistrationByPatient(c echo.Context) error {
 	c.Logger().Debug("hello world")
 	return c.JSON(http.StatusCreated, api.Return("ok", nil))
 }
 
-// DeleteRegistrationByPatient
-// @Summary delete a registration by its ID
-// @Tags patient
-// @Description return state
-// @Param registration_id path uint true "Registration ID"
+// GetRegistrationByDoctor
+// @Summary get a registration by its ID (doctor view)
+// @Tags doctor
+// @Description return a registration details by its ID
+// @Param RegistrationID body uint true "registration's ID"
 // @Produce json
-// @Success 200 {string} api.ReturnedData{"delete success"}
-// @Router /patient/registration/delete/{registration_id} [GET]
-func (h *RegistrationHandler) DeleteRegistrationByPatient(c echo.Context) error {
-	//verify identity
+// @Success 200 {object} api.ReturnedData{data=Registration}
+// @Router /doctor/registration [POST]
+func (h *RegistrationHandler) GetRegistrationByDoctor(c echo.Context) error {
 	c.Logger().Debug("hello world")
 	return c.JSON(http.StatusCreated, api.Return("ok", nil))
 }
 
 // UpdateRegistrationStatus
 // @Summary update registration status
-// @Tags doctor
-// @Description doctor change registration status
-// @Param registration_id body uint true "Registration ID"
-// @Param status body string true "Next registration's status"
+// @Tags both doctor & patient
+// @Description update registration status
+// @Param RegistrationID body uint true "Registration ID"
```

Comment:
Ditto

Reply:
This is marked as outdated, still have problem with my commit?